src/platforms/esp32/main/ledc_nifs.c

```diff
@@ -0,0 +1,503 @@
+/***************************************************************************
+ *   Copyright 2020 by Fred Dushin <fred@dushin.net>                       *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU Lesser General Public License as        *
+ *   published by the Free Software Foundation; either version 2 of the    *
+ *   License, or (at your option) any later version.                       *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA .        *
+ ***************************************************************************/
+
+#define _GNU_SOURCE
+
+#include "atom.h"
+#include "defaultatoms.h"
+#include "platform_defaultatoms.h"
+#include "nifs.h"
+#include "memory.h"
+#include "term.h"
+#include "interop.h"
+
+#include <stdlib.h>
+#include <driver/ledc.h>
+
+//#define ENABLE_TRACE
+#include "trace.h"
+
```

Comment:
I think I will define these macros in somekind of header file, to avoid further code duplication.

Reply:
Moved to nifs.h

Reply:
I put them in nits.h.  Let me know if there is a better place.